src/sql/src/plan/statement/dml.rs

```diff
@@ -421,52 +459,124 @@ pub fn plan_subscribe(
         options,
         as_of,
         up_to,
+        output,
     }: SubscribeStatement<Aug>,
     copy_to: Option<CopyFormat>,
 ) -> Result<Plan, PlanError> {
-    let from = match relation {
+    let (from, desc, order_by, project) = match relation {
         SubscribeRelation::Name(name) => {
             let entry = scx.get_item_by_resolved_name(&name)?;
-            match entry.item_type() {
-                CatalogItemType::Table
-                | CatalogItemType::Source
-                | CatalogItemType::View
-                | CatalogItemType::MaterializedView => SubscribeFrom::Id(entry.id()),
-                CatalogItemType::Func
-                | CatalogItemType::Index
-                | CatalogItemType::Sink
-                | CatalogItemType::Type
-                | CatalogItemType::Secret
-                | CatalogItemType::Connection => sql_bail!(
+            let desc = match entry.desc(&scx.catalog.resolve_full_name(entry.name())) {
+                Ok(desc) => desc,
+                Err(..) => sql_bail!(
                     "'{}' cannot be subscribed to because it is a {}",
                     name.full_name_str(),
                     entry.item_type(),
                 ),
+            };
+
+            // In constrast to a query, the order by cannot be very fancy, it's just column names
+            let mut order_by_col = Vec::new();
+            if let Some(SubscribeOutput::WithinTimestampOrderBy { order_by }) = &output {
+                for mut obe in order_by.clone() {
+                    let idx = match obe.expr.take() {
+                        mz_sql_parser::ast::Expr::Identifier(mut id) => {
+                            let id = normalize::column_name(id.pop().expect("shouldn't be empty"));
```

Comment:
I don't think it's safe to ignore the other elements if id. Can `plan_order_by_exprs` be used instead?

Reply:
yep, changed